app/controllers/partners/referrals_controller.rb

```diff
@@ -0,0 +1,61 @@
+module Partners
+    class ReferralsController < ApplicationController
+    include PromosHelper
+
+    def index
+        render "referrals"
+    end
+
+    def data
+        data = {}
+        unassigned_codes = current_publisher.promo_registrations.where(promo_campaign_id: nil)
+        campaigns = []
+    
+        current_publisher.promo_campaigns.each do |promo_campaign|
+        promo_registrations = current_publisher.promo_registrations.where(promo_campaign_id: promo_campaign.id)
+        campaigns.push(
```

Comment:
Instead of building this here, can we move this data aggregation to the model, similar to how [`Report.rb`](https://github.com/brave-intl/publishers/blob/f3b4ebd5d9646f2c7172b2bad8776d022a7eca60/app/models/report.rb#L8) does it?

It cleans up the controller a bit.